static/app/utils/discover/fields.tsx

```diff
@@ -616,16 +621,88 @@ export function getMeasurementSlug(field: string): string | null {
   return null;
 }
 
-const AGGREGATE_PATTERN = /^([^\(]+)\((.*?)(?:\s*,\s*(.*))?\)$/;
+const AGGREGATE_PATTERN = /^([^\(]+)\((.*)?\)$/;
 
 export function getAggregateArg(field: string): string | null {
+  // only returns the first argument if field is an aggregate
+  const result = parseFunction(field);
+
+  if (result && result.arguments.length > 0) {
+    return result.arguments[0];
+  }
+
+  return null;
+}
+
+export function parseFunction(field: string): ParsedFunction | null {
   const results = field.match(AGGREGATE_PATTERN);
-  if (results && results.length >= 3) {
-    return results[2];
+  if (results && results.length === 3) {
+    return {
+      name: results[1],
+      arguments: parseArguments(results[1], results[2]),
+    };
   }
+
   return null;
 }
 
+export function parseArguments(functionText: string, columnText: string): string[] {
+  // Some functions take a quoted string for their arguments that may contain commas
+  // This function attempts to be identical with the similarly named parse_arguments
+  // found in src/sentry/search/events/fields.py
+  if (
+    (functionText !== 'to_other' && functionText !== 'count_if') ||
+    columnText.length === 0
+  ) {
+    return columnText ? columnText.split(',').map(result => result.trim()) : [];
+  }
+
+  const args: string[] = [];
+
+  let quoted = false;
+  let escaped = false;
+
+  let i: number = 0;
+  let j: number = 0;
+
+  while (j < columnText.length) {
+    if (i === j && columnText[j] === '"') {
+      // when we see a quote at the beginning of
+      // an argument, then this is a quoted string
+      quoted = true;
+    } else if (quoted && !escaped && columnText[j] === '\\') {
+      // when we see a slash inside a quoted string,
+      // the next character is an escape character
+      escaped = true;
+    } else if (quoted && !escaped && columnText[j] === '"') {
+      // when we see a non-escaped quote while inside
+      // of a quoted string, we should end it
+      quoted = false;
+    } else if (quoted && escaped) {
+      // when we are inside a quoted string and have
+      // begun an escape character, we should end it
+      escaped = false;
+    } else if (quoted && columnText[j] === ',') {
+      // when we are inside a quoted string and see
+      // a comma, it should not be considered an
+      // argument separator
+    } else if (columnText[j] === ',') {
+      // when we see a comma outside of a quoted string
+      // it is an argument separator
+      args.push(columnText.substr(i, j - i).trim());
```

Comment:
Nit: A bit less roundabout and is closer to what the backend does

```suggestion
      args.push(columnText.substring(i, j).trim());
```

Reply:
... dunno why I used substr its deprecated even, will update :+1: